### PR TITLE
Update the Go version to 1.22

### PR DESCRIPTION
**Pull request by robertsirc**
This change is to update the go version from 1.21 to 1.22 and the golangci-lint from 1.55 to 1.56.

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -19,4 +19,4 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc #pin@3.7.0
         with:
-          version: v1.55
+          version: v1.56
```
Comment (mattfarina) [inline .github/workflows/golangci-lint.yml:22]: Earlier in this file there is a line that reads

```
go-version: "1.21"
```

This needs to be updated to 1.22.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module helm.sh/helm/v3
 
-go 1.21
+go 1.22
```
Comment (mattfarina) [inline go.mod:3]: It appears the [codeql](https://codeql.github.com/) (security scanner) does not support Go 1.22 being declared in the `go.mod` file. The change has been made but not yet released.

Can you undo this change for now? We can make it later.

**Review by mattfarina — COMMENTED** [review]
I think I found why the tests are failing...

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (robertsirc) [reply .github/workflows/golangci-lint.yml:22]: Silly mistake sorry.

File: go.mod (hunk 2 above)
Comment (robertsirc) [reply go.mod:3]: Reverting this now.

**Comment by robertsirc** [comment]
@mattfarina should I abandon this after our discussion?

**Pull request closed without merging**